test/common/stats/symbol_table_fuzz_test.cc

```diff
@@ -0,0 +1,43 @@
+#include "common/common/assert.h"
+#include "common/common/base64.h"
+#include "common/stats/symbol_table_impl.h"
+
+#include "test/fuzz/fuzz_runner.h"
+
+namespace Envoy {
+namespace Stats {
+namespace Fuzz {
+
+// Fuzzer for symbol tables.
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  FuzzedDataProvider provider(buf, len);
+  SymbolTableImpl symbol_table;
+  StatNamePool pool(symbol_table);
+  bool provider_empty = provider.remaining_bytes() == 0;
+
+  const uint64_t ResetPoolAfterN = 10000;
+  uint64_t num_in_pool = 0;
+
+  while (!provider_empty) {
+    if (num_in_pool == ResetPoolAfterN) {
+      pool.clear();
+      num_in_pool = 1;
+      ENVOY_LOG_MISC(error, "Note: Resetting pool");
```

Comment:
Should we just `break` at this point? It seems logically equivalent to break and rely on later fuzz runs without the prefixed operations that led to the limit.

Reply:
I agree, alternatively you could consider limiting the number of statnames added to the pool by a chosen integral from the provider:
```
const int num_iterations = fdp.ConsumeIntegralInRange(0, ResetPoolAfterN);
for (int i = 0; i < num_iterations; i++) {
  std::string next_data = std::string next_data = provider.ConsumeRandomLengthString(provider.remaining_bytes());
  ...
}
```

Reply:
I'm not sure, I could buy that. I'm still new to this library; it looks like it might be doing something introspective with debug symbols and maybe the more it runs the more it learns how to find corner cases?

I mainly wanted to clear the pool to bound the memory consumption, but I thought the fuzzer should continue to run. But I could just do say 10 rounds of 10k encode/decode.

Reply:
Let's say you fuzz [1-10k, 1-10k, 1-10k] with a stream of data `abcd123...something...foo`. This is exactly the same as running the fuzzer three times [1-10k] with distinct `abcd123...`, `something...`, `foo...` fuzz inputs. I don't think you get any advantage by doing the reset within the test, and it just adds complexity which might confuse the fuzzer.

Reply:
Simplified to let the fuzzer do its thing